### PR TITLE
Add role support to MI CLI

### PR DESCRIPTION
**Pull request by GDLMadushanka**
Fixes wso2/micro-integrator/issues/2571

### CODE REVIEW

**Comment by wasuradananjith** [comment]
Shall we generate the docs as shown in [1] and commit those files to this PR? (After generating the docs there will be changes apictl_set.md [2] file. Make sure **not to commit** those changes)

[1] https://github.com/wso2/product-apim-tooling/blob/master/import-export-cli/README.md?plain=1#L35
[2] https://github.com/wso2/product-apim-tooling/blob/master/import-export-cli/docs/apictl_set.md

File: import-export-cli/mi/impl/common.go [hunk 1]
```diff
@@ -165,6 +165,23 @@ func invokeDELETERequestWithRetry(url string, env string) (*resty.Response, erro
 	})
 }
 
+func invokeDELETERequestWithRetryAndParams(url string, env string, params map[string]string) (*resty.Response, error) {
```
Comment (wasuradananjith) [inline import-export-cli/mi/impl/common.go:168]:
```suggestion
func invokeDELETERequestWithRetryAndParams(url, env string, params map[string]string) (*resty.Response, error) {
```
Shall we refactor the method signature by grouping the string arguments together, as shown above, to reduce the number of characters per line?

File: import-export-cli/mi/impl/constants.go [hunk 2]
```diff
@@ -41,3 +41,4 @@ const monthHeader = "MONTH"
 const yearHeader = "YEAR"
 const transactionCountHeader = "TRANSACTION COUNT"
 const userIDHeader = "USER ID"
+const RoleHeader = "ROLE"
```
Comment (wasuradananjith) [inline import-export-cli/mi/impl/constants.go:44]:
```suggestion
const roleHeader = "ROLE"
```
Any reason why we made this RoleHeader variable start from uppercase?
"If the functions with names that start with an uppercase letter will be exported to other packages. If the function name starts with a lowercase letter, it won't be exported to other packages, but you can call this function within the same package." [1]

[1] https://www.golangprograms.com/naming-conventions-for-golang-functions.html

File: import-export-cli/cmd/mi/update/user.go [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+func handleupdateUserCmdArguments(args []string) {
+	printUpdateCmdVerboseLog(miUtils.GetTrimmedCmdLiteral(updateUserCmdLiteral))
+	credentials.HandleMissingCredentials(updateUserCmdEnvironment)
+	startConsoleToUpdateUser(args[0])
+}
+
+func executeUpdateUser(userName, domain string, addedRoles, removedRoles []string) {
+	resp, err := impl.UpdateMIUser(updateUserCmdEnvironment,userName, domain, addedRoles, removedRoles)
```
Comment (wasuradananjith) [inline import-export-cli/cmd/mi/update/user.go:69]:
```suggestion
	resp, err := impl.UpdateMIUser(updateUserCmdEnvironment ,userName, domain, addedRoles, removedRoles)
```
shall we format all the changes accordingly? (You can follow [1] under the topic setting up VS code)

[1] https://wasuradananjith.medium.com/significance-of-go-part-2-play-with-vs-code-and-introducing-go-packages-and-modules-8d39da7318a8

File: import-export-cli/cmd/mi/delete/role.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	Args:    cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		handledeleteRoleCmdArguments(args)
+	},
+}
+
+func init() {
+	DeleteCmd.AddCommand(deleteRoleCmd)
+	deleteRoleCmd.Flags().StringVarP(&deleteRoleCmdDomain, "domain", "d", "", "select the domain of the role")
+	deleteRoleCmd.Flags().StringVarP(&deleteRoleCmdEnvironment, "environment", "e", "", "Environment of the micro integrator from which a role should be deleted")
```
Comment (wasuradananjith) [inline import-export-cli/cmd/mi/delete/role.go:59]:
```suggestion
	deleteRoleCmd.Flags().StringVarP(&deleteRoleCmdEnvironment, "environment", "e", "", "Environment of the Micro Integrator from which a role should be deleted")
```
shall we have one convention in all the places? For example, to capitalize the MI or not?

File: import-export-cli/cmd/mi/update/user.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+var updateUserCmdEnvironment string
+
+const updateUserCmdLiteral = "user [user-name]"
+const updateUserCmdShortDesc = "Update roles of a user in a Micro Integrator"
+
+const updateUserCmdLongDesc = "Update the roles of a user named [user-name] specified by the command line arguments in a Micro Integrator in the environment specified by the flag --environment, -e"
+
+var updateUserCmdExamples = "To update the roles\n" +
+	"  " + utils.ProjectName + " " + utils.MiCmdLiteral + " " + updateCmdLiteral + " " + miUtils.GetTrimmedCmdLiteral(updateUserCmdLiteral) + " [user-name] DEBUG -e dev\n" +
```
Comment (SanojPunchihewa) [inline import-export-cli/cmd/mi/update/user.go:42]: Is this added mistakenly?
```suggestion
	"  " + utils.ProjectName + " " + utils.MiCmdLiteral + " " + updateCmdLiteral + " " + miUtils.GetTrimmedCmdLiteral(updateUserCmdLiteral) + " [user-name] -e dev\n" +
```

File: import-export-cli/mi/impl/common.go (hunk 1 above)
Comment (GDLMadushanka) [reply import-export-cli/mi/impl/common.go]: fixed in 84dfebb1a9d2419c951bfbb295afe6ccd5c8dfe2

File: import-export-cli/mi/impl/constants.go (hunk 2 above)
Comment (GDLMadushanka) [reply import-export-cli/mi/impl/constants.go]: fixed in 84dfebb1a9d2419c951bfbb295afe6ccd5c8dfe2

File: import-export-cli/cmd/mi/delete/role.go (hunk 4 above)
Comment (GDLMadushanka) [reply import-export-cli/cmd/mi/delete/role.go]: fixed in 84dfebb1a9d2419c951bfbb295afe6ccd5c8dfe2

File: import-export-cli/cmd/mi/update/user.go (hunk 5 above)
Comment (GDLMadushanka) [reply import-export-cli/cmd/mi/update/user.go]: fixed in 84dfebb1a9d2419c951bfbb295afe6ccd5c8dfe2

**Review by wasuradananjith — APPROVED** [review]

**Pull request merged**